Content.Shared/Access/Systems/ActivatableUIRequiresAccessSystem.cs

```diff
@@ -0,0 +1,30 @@
+using Content.Shared.Popups;
+using Content.Shared.UserInterface;
+using Content.Shared.Access.Components;
+
+namespace Content.Shared.Access.Systems;
+public sealed class ActivatableUIRequiresAccessSystem : EntitySystem
+{
+    [Dependency] private readonly AccessReaderSystem _access = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ActivatableUIRequiresAccessComponent, ActivatableUIOpenAttemptEvent>(OnUIOpenAttempt);
+    }
+
+    private void OnUIOpenAttempt(Entity<ActivatableUIRequiresAccessComponent> activatableUI, ref ActivatableUIOpenAttemptEvent args)
+    {
+        if (args.Cancelled)
+            return;
+
+        if (!_access.IsAllowed(args.User, activatableUI))
+        {
+            args.Cancel();
+            _popup.PopupEntity(Loc.GetString("lock-comp-has-user-access-fail"), activatableUI, args.User);
+        }
```

Comment:
Would be better if the event had a field for popups instead, because if a caller wants this to fail silently there's no way to do so.

Also needs to be popupclient because this will spam when we get clientside UIs.